src/test/java/StringManipulationTest.java

```diff
@@ -50,100 +69,194 @@ public void testRemoveNthCharacter1() {
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("What happens if n is 1?");
+        assertEquals("                       ", manipulatedstring.removeNthCharacter(1, true));
     }
 
+    // Tests whether method throws error when n = 0
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("An exception should be thrown if n = 0");
+        try {
+            manipulatedstring.removeNthCharacter(0, false);
+            fail("No exception thrown for n=0");
+        } catch (IllegalArgumentException e) {
+            assert true;
+        }
     }
 
+    // Tests whether method throws error when n > length
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("An exception should ALSO be thrown if n > length");
+        try {
+            manipulatedstring.removeNthCharacter(100, false);
+            fail("No exception thrown for n > length");
+        } catch (IndexOutOfBoundsException e) {
+            assert true;
+        }
     }
 
     @Test
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+        String[] sStings = manipulatedstring.getSubStrings(3, 4);
 
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Lets see what happens when we pass in a string with lots and lots of words");
+        String[] substrings = manipulatedstring.getSubStrings(4, 10);
+        assertEquals(substrings[0], "happens");
+        assertEquals(substrings[3], "pass");
+        assertEquals(substrings[6], "string");
     }
+
+    // Tests a string with newline characters
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Trying out some\nnewline characters");
+        String[] subStrings = manipulatedstring.getSubStrings(3, 4);
+        assertEquals(subStrings[0], "some");
+        assertEquals(subStrings[1], "newline");
     }
+
+    // Tests a string with numbers and symbols
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring
+                .setString("Special characters! Punctuation, symbols !@#$ and numbers 91283... Will it matter?");
+        String[] substrings = manipulatedstring.getSubStrings(2, 10);
+        assertEquals(substrings[0], "characters!");
+        assertEquals(substrings[3], "!@#$");
+        assertEquals(substrings[6], "91283...");
     }
+
+    // Test whether exceptions are thrown when startWord > endWord or when one of the variables is < 0
     @Test
```

Comment:
In Java 8+, you can use Lamba expressions with JUnit to handle exceptions. Here's a resource that shows how this is done: https://blog.codeleak.pl/2014/07/junit-testing-exception-with-java-8-and-lambda-expressions.html

Reply:
Thanks! I've been meaning to work more with lambda expressions. I'll be sure to check out your implementation as well.